features/BLT001_Correct-use-of-operation-type-attributes-for-doors.feature

```diff
@@ -0,0 +1,28 @@
+@implementer-agreement
+@BLT
+@version1
+@E00020
+
+Feature: BLT001 - Correct use of operation type attributes for doors
+
+  The rule verifies that attribuet UserDefinedOperationType is provided only when 
+  the value of the attribute OperationType is set to USERDEFINED.
+
+
+Background:
+  Given an IfcDoor
+
+
+Scenario: Correct values for OperationType and UserDefinedOperationType
+
+  Given UserDefinedOperationType = not empty
```

Comment:
```suggestion
  Given UserDefinedOperationType is not empty
```

Reply:
I did this initially. However there is a caveat with the following lines 
```
    pred = operator.eq
    if value == 'empty':
        value = ()
    elif value == 'not empty':
        value = ()
```
When we say 'is not empty', `value` will be equal to '`empty` and not` 'not empty'` (the not part will is then parsed in the `comparison_to_op` part).

Reply:
Ok, sounds like there is a good reason to leave as is so we can drop my suggestion.

Reply:
I agree with you, it's more if you can intuitively write it down as` is not empty`

Tried to solve it, but there is an error later in the code as well. I don't find an easy standardized solution that covers all the cases that we have in our rules. It's probably not preferable at the moment to spend too much time on what is actually more a technical depth than writing rules.
In the meantime .. I left a note in case someone trips over this issue.